src/renderers/dom/fiber/ReactDOMFiber.js

```diff
@@ -61,9 +70,29 @@ let selectionInformation : ?mixed = null;
 
 var DOMRenderer = ReactFiberReconciler({
 
-  getChildHostContext(parentHostContext : string | null, type : string) {
-    const parentNamespace = parentHostContext;
-    return getChildNamespace(parentNamespace, type);
+  getRootHostContext(rootContainerInstance) {
+    const type = rootContainerInstance.tagName.toLowerCase();
+    if (__DEV__) {
+      const namespace = getChildNamespace(null, type);
+      const isMountingIntoDocument = rootContainerInstance.ownerDocument.documentElement === rootContainerInstance;
+      const ancestorInfo = getChildAncestorInfo(null, isMountingIntoDocument ? '#document' : type);
```

Comment:
Good catch. It doesn't (in Stack neither), and this is a bug. Because that should be valid: https://jsfiddle.net/q7e61jpd/.